src/run.rs

```diff
@@ -1,38 +1,47 @@
 use chrono::prelude::*;
 use clap::ArgMatches;
 
-use crate::util::{
-    file::{cargo_path, day_path, download_input_file},
-    get_year,
-    submit::{self, get_submit_day},
+use crate::{
+    error::AocError,
+    util::{
+        file::{cargo_path, day_path, download_input_file},
+        get_year,
+        submit::{self, get_submit_day},
+    },
 };
 
-pub fn get_day(matches: &ArgMatches) -> Result<u32, std::num::ParseIntError>
+pub fn get_day(matches: &ArgMatches) -> Result<u32, AocError>
 {
-    matches.get_one::<String>("day").unwrap().parse()
+    let day = matches
+        .get_one::<String>("day")
+        .ok_or(AocError::ArgMatches)?
+        .parse::<u32>()
+        .map_err(|_| AocError::ParseNum)?;
+    if !(1..=25).contains(&day)
+    {
+        Err(AocError::InvalidRunDay)
+    }
+    else
+    {
+        Ok(day)
+    }
 }
 
-async fn get_input_file(matches: &ArgMatches) -> Result<String, Box<dyn std::error::Error>>
```

Comment:
Aah good catch, I originally wanted this function to additionally download the input file, but I decided against it. The async stems from that